parler_tts/modeling_parler_tts.py

```diff
@@ -3387,7 +3387,8 @@ def generate(
                 )
 
         # build the delay pattern mask for offsetting each codebook prediction by 1 (this behaviour is specific to Parler-TTS)
-        input_ids, decoder_delay_pattern_mask = self.decoder.build_delay_pattern_mask(
+        # but don't overwrite the input_ids tensor with the delay pattern mask. We perform that later
+        _, decoder_delay_pattern_mask = self.decoder.build_delay_pattern_mask(
```

Comment:
As pointed out, this is a redundant operation that has no impact on the results!

Reply:
Hmm, I think this line does indeed change the results when using enrolled tokens. Perhaps your setup is working because it is slightly different as you've described below. I shall try this and get back to you

Reply:
Ok, so my testing shows that this fix is required to get the right audio when doing the enrolment. Here is an example audio file generated with and without the fix:
[audio.zip](https://github.com/user-attachments/files/17155118/audio.zip)

Reply:
I can confirm that, overwriting input_ids with delay pattern mask leads to noisy audio generation. Although, passing input_values to generate() changes some behavior and the prompt (or prompt_input_ids) doesn't seem to be included in the audio generated.